src/parse-jobs.js

```diff
@@ -3,106 +3,91 @@ const cheerio = require('cheerio');
 const Entities = require('html-entities').AllHtmlEntities;
 
 const { log } = Apify.utils;
-const { BASE_URL, REQUEST_HEADERS } = require('./consts');
 
-const parseJobs = async (searchResults, proxyUrl) => {
+const parseJobs = async ({ request, session }, proxyConfiguration) => {
+    const { item } = request.userData;
     // encoding-decoding html entities
     // used to get jobDetails from JSON-LD instead of page content
     const entities = new Entities();
+    const rq = await Apify.utils.requestAsBrowser({
+        url: request.url,
+        proxyUrl: proxyConfiguration.newUrl(session.id),
+    });
+    // GETTING JSON WITH THE NEEDED INFO
+    const $ = cheerio.load(rq.body);
+    // GETTING INFO FROM THE PAGE
+    let clearDetails = $('#JobDescriptionContainer').text().trim(); // but no artifacts from html decoding here
+    let script = $('#JobView script').html();
+    try {
+        script = script.replace('window.appCache=', '').slice(0, -1);
+    } catch (e) {
+        log.debug('Error on getting JSON', { message: e.message, stack: e.stack });
+        throw new Error('The page didnt load properly, will try again...');
+    }
+    const jsonCompanyInfo = JSON.parse(script);        
 
-    // global variable for loaded cheerio content to keep jQuery-alike syntax
-    let $;
-
-    let rawdata;
-    let json;
-
-    const requestList = await Apify.openRequestList(
-        'LIST3',
-        searchResults.map((x) => ({ url: x.url, uniqueKey: x.id.toString() })),
-    );
-
-    // keep parsed details from company overview to avoid extra calls
-    const companyDetails = {};
-
-    const crawler = new Apify.BasicCrawler({
-        requestList,
-        handleRequestFunction: async ({ request }) => {
-            log.info(`job GET ${request.url}`);
-            const rq = await Apify.utils.requestAsBrowser({
-                url: request.url,
-                proxyUrl,
-                ...REQUEST_HEADERS,
-            });
-            $ = cheerio.load(rq.body);
-            rawdata = $('script[type="application/ld+json"]').html();
-            const cleanstr = rawdata.replace(/\s+/g, ' ').trim();
-            json = JSON.parse(cleanstr);
-            const updatedItem = searchResults.find((x) => x.id === parseInt(request.uniqueKey, 10));
-            if (!updatedItem) {
-                log.error(`Not found job listing id ${request.uniqueKey} in search results`);
-                return;
-            }
+    if (!clearDetails) {
+        // so for now second option will be used as jobDetails, decoding below was for json.description
+        clearDetails = jsonCompanyInfo.initialState.jlData.job.description; // html encoded, decoding is not 99% accurate
```

Comment:
last one `TypeError: Cannot read property 'description' of undefined`. either use nullish coalescing and optional chaining to an empty string or place this inside the try/catch below